File: CSharp/Behavioral/Mediator/Example2/Show/Show/TerminalExpression.cs
namespace Show
{
    public class TerminalExpression: Expression
    {
        private string data;

        public TerminalExpression(string data) {
            this.data = data;
        }
        
        public bool Interpret(string context)
        {
            return context.Contains(data);
        }
    }
}

File: Language/JavaScript/Behavioral/Interpreter/Example1/TerminalExpression.js
// TerminalExpression.js

const AbstractExpression = require('./AbstractExpression');

// TerminalExpression: implements an interpret method for terminal symbols
class TerminalExpression extends AbstractExpression {
    constructor(data) {
        super();
        this.data = data; // store terminal symbol
    }

    interpret(context) {
        // check if context contains the terminal data
        return context.includes(this.data);
    }
}

module.exports = TerminalExpression;
